### PR TITLE
Class to create temporary SSH keys.

### PR DESCRIPTION
**Pull request by boumenot**

### CODE REVIEW

File: packer/builder/azure/arm/openssh_key_pair_test.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+// See License.txt in the project root for license information.
+
+package arm
+
+import (
+	"testing"
+	"golang.org/x/crypto/ssh"
+)
+
+func TestAuthorizedKeyShouldBeParse(t *testing.T) {
```
Comment (paulmey) [inline packer/builder/azure/arm/openssh_key_pair_test.go]: ...ShouldBeParse -> ...ShouldParse for consistency?

File: packer/builder/azure/arm/openssh_key_pair.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+
+func (s *OpenSshKeyPair) AuthorizedKey() string {
+	b := bytes.NewBufferString("")
+	b.WriteString(s.publicKey.Type())
+	b.WriteString(" ")
+	b.WriteString(base64.StdEncoding.EncodeToString(s.publicKey.Marshal()))
+	b.WriteString(" ")
+	b.WriteString(fmt.Sprintf("packer Azure Deployment@%s", time.Now().Format(time.RFC3339)))
+	return b.String()
```
Comment (paulmey) [inline packer/builder/azure/arm/openssh_key_pair.go]: might as well put all of this just in an encompassing `fmt.Sprintf()`? IMHO that makes it a bit easier to read.

**Comment by paulmey** [comment]
LGTM, just a nit in the test naming for consistency

File: packer/builder/azure/arm/openssh_key_pair_test.go (hunk 1 above)
Comment (boumenot) [inline packer/builder/azure/arm/openssh_key_pair_test.go]: Fixed.

File: packer/builder/azure/arm/openssh_key_pair.go (hunk 2 above)
Comment (boumenot) [inline packer/builder/azure/arm/openssh_key_pair.go]: Agreed.

**Pull request merged**